osu.Game/Screens/Edit/Screens/Compose/Layers/MaskSelection.cs

```diff
@@ -63,39 +63,23 @@ private void load(OsuColour colours)
 
         #region User Input Handling
 
-        // Only handle input on selectable or selected masks
-        public override bool ReceiveMouseInputAt(Vector2 screenSpacePos) => selectableMasks.Reverse().Concat(selectedMasks).Any(m => m.ReceiveMouseInputAt(screenSpacePos));
+        public override bool ReceiveMouseInputAt(Vector2 screenSpacePos) => true;
 
-        protected override bool OnMouseDown(InputState state, MouseDownEventArgs args)
-        {
-            // If masks are overlapping, make sure we don't change the selection if the overlapped portion is pressed
-            if (selectedMasks.Any(m => m.ReceiveMouseInputAt(state.Mouse.NativeState.Position)))
-                return true;
+        protected override bool OnMouseDown(InputState state, MouseDownEventArgs args) => handleInput(state);
 
-            DeselectAll();
-            selectableMasks.Reverse().First(m => m.ReceiveMouseInputAt(state.Mouse.NativeState.Position)).Select();
+        protected override bool OnDragStart(InputState state) => handleInput(state);

```

Comment:
Because drag is delayed by a number of pixels, this will cause problems if you drag out from the border of a mask (no masks will be receiving mouse input).